documentation/specs/Select.md

```diff
@@ -0,0 +1,216 @@
+
+### Features
+
+- Setting the `size` property also sets size for iconAtStart: The size values map to Easy UI's token sizes for icons.
+- When `errorText` is supplied with `validationState="invalid"`, `helperText` is overriden.
+- `isLabelVisuallyHidden` property can be used to visually hide the label.
+- Supports disabled state.
+
+### Risks and Challenges
+
+- Ensuring dropdown doesn't inadvertently get cut off.
+- Maintaining an approachable and familiar API; similar to existing Easy UI form related components.
+- Ensuring the component remains accessible when label is visually hidden.
+
+### Prior Art
+
+- [Paste `<Select />`](https://paste.twilio.design/components/select)
+- [Polaris `<Select />`](https://polaris.shopify.com/components/selection-and-input/select)
+- [Spectrum `<Picker />`](https://react-spectrum.adobe.com/react-spectrum/Picker.html)
+
+---
+
+## Design
+
+### API
+
+```ts
+import type { AriaSelectProps } from "react-aria";
+
+export type SelectSize = "sm" | "md" | "lg";
+export type ValidationState = "valid" | "invalid";
+
+export type SelectProps = AriaSelectProps & {
+  /**
+   * Visually hides the label, but keeps it accessible.
+   * @default false
+   */
+  isLabelVisuallyHidden?: boolean;
+  /**
+   * Whether the select field is disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether user input is required on the select field before form submission.
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * Whether the select field should display its "valid" or "invalid" visual styling.
+   * @default valid
+   */
+  validationState?: ValidationState;
+  /**
+   * Label text displays with emphasis.
+   * @default false
+   */
+  isLabelEmphasized?: boolean;
+  /**
+   * Whether the element should receive focus on render.
+   * @default false
+   */
+  autoFocus?: boolean;
+  /**
+   * Size affects the overall size of the select field, but it also influences
+   * the size of iconAtStart.
+   * @default md
+   */
+  size?: SelectSize;
+  /** The content to display as the label. */
+  label: ReactNode;
+  /** Error text that appears below select field. */
+  errorText?: ReactNode;
+  /** Helper text that appears below select field. */
+  helperText?: ReactNode;
+  /** Left aligned icon on select field. */
+  iconAtStart?: IconSymbol;
+  /** Method that is called when the open state of the select field changes. */
+  onOpenChange?: (isOpen: boolean) => void;
+  /** Sets the open state of the select field. */
+  isOpen?: boolean;
+  /** The currently selected key in the collection (controlled). */
+  selectedKey?: Key | null;
+  /** The initial selected key in the collection (uncontrolled). */
+  defaultSelectedKey?: Key;
+  /** Handler that is called when the selection changes. */
+  onSelectionChange?: (key: Key) => any;
+  /** The contents of the collection. */
+  children: CollectionChildren<T>;
+  /** Item objects in the collection. */
+  items?: Iterable<T>;
```

Comment:
**thought:** i feel iffy on having these two approaches for defining items. reading through the stately docs, i understand that using `items` gives you performance benefits, but i think it's confusing for consumers to have to decide when to use what.

i'm personally of the opinion that we should only expose one paradigm in our components based on what we feel makes the most sense for the component. there may be some, like a table, that needs to be concerned with performance and so we need to use `items`, but i think a select shouldn't need to. in which case, i would be a proponent of only having one option for adding items using the `children` prop.

so for dynamic data the consumer can use the same approach as they would with static data and map through the array elements:

```tsx
<ListBox>
  {animals.map(item =>
    <Item key={item.name}>{item.name}</Item>
  )}
</ListBox>
```

what do you think?